fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java

```diff
@@ -0,0 +1,180 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.binlog;
+
+import com.starrocks.alter.SchemaChangeHandler;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Partition;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.util.PropertyAnalyzer;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.thrift.TTabletMetaType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class BinlogManager {
+
+    private static final Logger LOG = LogManager.getLogger(BinlogManager.class);
+
+    SchemaChangeHandler schemaChangeHandler;
+
+    private Map<Long, List<Long>> binlogTxnIdToTableIds;
+
+    // When enable binlog, there may be concurrent imports
+    // record the map between all txnIds of concurrently imports
+    // and binlogTxnId
+    private Map<Long, Set<Long>> transcationidToBinlogTxnId;
+
+
+    private ReentrantReadWriteLock binlogLock = new ReentrantReadWriteLock(true);
+
+    public BinlogManager() {
+        binlogTxnIdToTableIds = new HashMap<>();
+        transcationidToBinlogTxnId = new HashMap<>();
+        schemaChangeHandler = GlobalStateMgr.getCurrentState().getSchemaChangeHandler();
+    }
+
+    public void addBinlogTxnIdToTableId(Long binlogTxnId, Long tableId) {
+        if (binlogTxnIdToTableIds.containsKey(binlogTxnId)) {
+            // should not happen, for the binlogTxnId can not be  duplicate
+            LOG.warn("the binlogTxnId has added to the binlog manager");
+            return;
+        }
+        binlogTxnIdToTableIds.putIfAbsent(binlogTxnId, new ArrayList<Long>()).add(tableId);
+    }
+
+    public boolean isBinlogWaitingTransactionId(Long transactionId) {
+        binlogLock.readLock().lock();
+        if (transcationidToBinlogTxnId.containsKey(transactionId)) {
+            return true;
+        }
+        binlogLock.readLock().unlock();
```

Comment:
emm. Use`try-finally` style to avoid leak the lock?

Reply:
ok